com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -704,16 +713,20 @@ public override void OnNetworkSpawn()
             {
                 TryCommitTransformToServer(m_Transform, m_CachedNetworkManager.LocalTime.Time);
             }
-            m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
 
-            // crucial we do this to reset the interpolators so that recycled objects when using a pool will
-            //  not have leftover interpolator state from the previous object
+            // We do this to reset the interpolators so that pooled NetworkObjects will initialize prior
+            // to assigning the current NetworkState -- helps prevent pooled objects from interpolating from the last state
             Initialize();
+
+            m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
         }
 
         public override void OnNetworkDespawn()
         {
+            // Reset the network state once despawned -- helps prevent pooled objects from interpolating from the last state
+            m_LocalAuthoritativeNetworkState.Reset();
```

Comment:
I keep coming back to thinking we need something more formal / explicit for these kinds of "extra cleanup to support pooling".  But for a spot fix this seems reasonable.